lib/puppet/util/windows/adsi.rb

```diff
@@ -120,20 +120,51 @@ def self.parse_name(name)
       return account, domain
     end
 
-    def native_user
-      @native_user ||= Puppet::Util::Windows::ADSI.connect(self.class.uri(*self.class.parse_name(@name)))
+    def get_sids(adsi_child_collection)
+      sids = []
+      adsi_child_collection.each do |m|
+        sids << Puppet::Util::Windows::SID.octet_string_to_sid_object(m.objectSID)
+      end
+
+      sids
     end
 
-    def sid
-      @sid ||= Puppet::Util::Windows::SID.octet_string_to_sid_object(native_user.objectSID)
+    def name_sid_hash(names)
+      return {} if names.nil? or names.empty?
+
+      sids = names.map do |name|
+        sid = Puppet::Util::Windows::SID.name_to_sid_object(name)
+        raise Puppet::Error.new( "Could not resolve name: #{name}" ) if !sid
+        [sid.to_s, sid]
+      end
+
+      Hash[ sids ]
     end
+  end
 
-    def self.uri(name, host = '.')
-      if sid_uri = Puppet::Util::Windows::ADSI.sid_uri_safe(name) then return sid_uri end
+  class User
+    extend Enumerable
+    extend Puppet::Util::Windows::ADSI::Shared
```

Comment:
Ah right... hmm.

It's too bad we haven't been better at marking things `@private`.  Realistically, I think if we don't have callers internally to Puppet that we're OK.... but... you're technically correct wrt semver.